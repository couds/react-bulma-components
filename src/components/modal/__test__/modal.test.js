import React from 'react';
import { mount } from 'enzyme';
import { JSDOM } from 'jsdom';
import { renderToString } from 'react-dom/server';
import Modal from '..';

describe('Modal component', () => {
  let window;
  let component;
  beforeEach(() => {
    // eslint-disable-next-line
    window = new JSDOM().window;
    global.window = window;
    global.document = window.document;
  });
  afterEach(() => {
    if (component && component.unmount) {
      component.unmount();
    }
    window = undefined;
    global.window = undefined;
    global.document = undefined;
  });
  it('Should Exist', () => {
    expect(Modal).toMatchSnapshot();
  });
  it("Should expose Content and Card's types", () => {
    expect(Modal.Content).toMatchSnapshot();
    expect(Modal.Card).toMatchSnapshot();
    expect(Modal.Card.Title).toMatchSnapshot();
    expect(Modal.Card.Head).toMatchSnapshot();
    expect(Modal.Card.Body).toMatchSnapshot();
    expect(Modal.Card.Foot).toMatchSnapshot();
  });
  it('Should render modal-card-head', () => {
    const onClose = jest.fn();
    component = mount(
      <Modal.Card onClose={onClose}>
        <Modal.Card.Head onClose={onClose}>
          <Modal.Card.Title>Modal Title</Modal.Card.Title>
        </Modal.Card.Head>
      </Modal.Card>,
    );
    component
      .find('.delete')
      .first()
      .simulate('click');
    expect(onClose).toHaveBeenCalledTimes(1);
  });
  it('Should open the modal', () => {
    const onClose = jest.fn(() => {
      component.setProps({ show: false });
    });
    component = mount(
      <Modal document={window.document} show={false} onClose={onClose}>
        <Modal.Card>
          <Modal.Card.Head>
            <Modal.Card.Title>Modal Title</Modal.Card.Title>
          </Modal.Card.Head>
        </Modal.Card>
      </Modal>,
    );
    expect(window.document.querySelector('div.modal.is-active')).toBeNull();
    component.setProps({ show: true });
    expect(
      window.document.querySelector('div.modal.is-active'),
    ).toMatchSnapshot();
  });
  it('Should close the modal', () => {
    const onClose = jest.fn(() => {
      component.setProps({ show: false });
    });
    component = mount(
      <Modal document={window.document} show onClose={onClose}>
        <Modal.Card>
          <Modal.Card.Head onClose={onClose}>
            <Modal.Card.Title>Modal Title</Modal.Card.Title>
          </Modal.Card.Head>
        </Modal.Card>
      </Modal>,
    );
    const modal = window.document.querySelector('div.modal.is-active');
    expect(modal).toMatchSnapshot();
    modal.querySelector('.delete').click();
    expect(onClose).toHaveBeenCalledTimes(1);
    expect(window.document.querySelector('div.modal.is-active')).toBeNull();
  });
  it('Should close the modal on ESC key press', () => {
    const onClose = jest.fn(() => {
      component.setProps({ show: false });
    });
    component = mount(
      <Modal document={window.document} show onClose={onClose}>
        <Modal.Card>
          <Modal.Card.Head>
            <Modal.Card.Title>Modal Title</Modal.Card.Title>
          </Modal.Card.Head>
        </Modal.Card>
      </Modal>,
    );
    const modal = window.document.querySelector('div.modal.is-active');
    expect(modal).toMatchSnapshot();
    const event = new window.KeyboardEvent('keydown', { keyCode: 27 });
    window.document.dispatchEvent(event);
    expect(onClose).toHaveBeenCalledTimes(1);
    expect(window.document.querySelector('div.modal.is-active')).toBeNull();
  });
  it('Should not close the modal on ESC key press', () => {
    const onClose = jest.fn(() => {
      component.setProps({ show: false });
    });
    component = mount(
      <Modal
        closeOnEsc={false}
        document={window.document}
        show
        onClose={onClose}
      >
        <Modal.Card>
          <Modal.Card.Head>
            <Modal.Card.Title>Modal Title</Modal.Card.Title>
          </Modal.Card.Head>
        </Modal.Card>
      </Modal>,
    );
    const modal = window.document.querySelector('div.modal.is-active');
    expect(modal).toMatchSnapshot();
    const event = new window.KeyboardEvent('keydown', { keyCode: 27 });
    window.document.dispatchEvent(event);
    expect(onClose).not.toHaveBeenCalled();
    expect(modal).not.toBe(null);
  });
  it('Should not close the modal on other than ESC key press', () => {
    const onClose = jest.fn(() => {
      component.setProps({ show: false });
    });
    component = mount(
      <Modal closeOnEsc document={window.document} show onClose={onClose}>
        <Modal.Card>
          <Modal.Card.Head>
            <Modal.Card.Title>Modal Title</Modal.Card.Title>
          </Modal.Card.Head>
          <Modal.Card.Body>Body</Modal.Card.Body>
          <Modal.Card.Foot>Footer</Modal.Card.Foot>
        </Modal.Card>
      </Modal>,
    );
    const modal = window.document.querySelector('div.modal.is-active');
    expect(modal).toMatchSnapshot();
    const event = new window.KeyboardEvent('keydown', { keyCode: 23 });
    window.document.dispatchEvent(event);
    expect(onClose).not.toHaveBeenCalled();
    expect(modal).not.toBe(null);
  });
  it('Should render any child type', () => {
    // eslint-disable-next-line no-console
    const onClose = jest.fn();
    component = mount(
      <Modal document={window.document} show onClose={onClose}>
        <div>CHILDREN</div>
        <div>CHILDREN</div>
      </Modal>,
    );
    expect(
      window.document.querySelector('div.modal.is-active'),
    ).toMatchSnapshot();
  });
  it('Should no try to reopen if other prop change', () => {
    const onClose = jest.fn();
    component = mount(
      <Modal document={window.document} show onClose={onClose}>
        <Modal.Content>Content</Modal.Content>
      </Modal>,
    );
    component.setProps({ randomProp: 'test' });
    expect(
      window.document.querySelector('div.modal.is-active'),
    ).toMatchSnapshot();
  });
  it('Should close the modal if clicked on background', () => {
    const onClose = jest.fn(() => {
      component.setProps({ show: false });
    });
    component = mount(
      <Modal document={window.document} show onClose={onClose} closeOnBlur>
        <Modal.Content>Content</Modal.Content>
      </Modal>,
    );
    const modal = window.document.querySelector('div.modal.is-active');
    expect(modal).toMatchSnapshot();
    modal.querySelector('div.modal-background').click();
    expect(onClose).toHaveBeenCalledTimes(1);
    expect(window.document.querySelector('div.modal.is-active')).toBeNull();
  });
  it('Should not close the modal if clicked on background', () => {
    const onClose = jest.fn(() => {
      component.setProps({ show: false });
    });
    component = mount(
      <Modal
        closeOnBlur={false}
        document={window.document}
        show
        onClose={onClose}
      >
        <Modal.Content>Content</Modal.Content>
      </Modal>,
    );
    const modal = window.document.querySelector('div.modal.is-active');
    expect(modal).toMatchSnapshot();
    modal.querySelector('div.modal-background').click();
    expect(onClose).not.toHaveBeenCalled();
    expect(
      window.document.querySelector('div.modal.is-active'),
    ).toMatchSnapshot();
  });
  it('Should not show close button', () => {
    const onClose = jest.fn(() => {
      component.setProps({ show: false });
    });
    component = mount(
      <Modal
        showClose={false}
        document={window.document}
        show
        onClose={onClose}
      >
        <Modal.Content>Content</Modal.Content>
      </Modal>,
    );
    const modal = window.document.querySelector('div.modal.is-active');
    expect(modal).toMatchSnapshot();
  });

  it('Should render empty because no document on scope', () => {
    component = renderToString(
      <Modal show onClose={() => {}}>
        <Modal.Content>Content</Modal.Content>
      </Modal>,
    );
    expect(component).toMatchSnapshot();
  });
});
