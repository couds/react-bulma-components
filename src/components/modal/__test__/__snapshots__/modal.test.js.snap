// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component Should Exist 1`] = `[Function]`;

exports[`Modal component Should close the modal 1`] = `
<div
  class="modal is-active"
>
  <div
    class="modal-background"
    role="presentation"
  />
  <div
    class="modal-card"
  >
    <header
      class="modal-card-head"
    >
      <p
        class="modal-card-title"
      >
        Modal Title
      </p>
      <a
        class="delete"
        href=""
      />
    </header>
  </div>
</div>
`;

exports[`Modal component Should close the modal if clicked on background 1`] = `
<div
  class="modal is-active"
>
  <div
    class="modal-background"
    role="presentation"
  />
  <div
    class="modal-content"
  >
    Content
  </div>
  <button
    aria-label="close"
    class="modal-close is-large"
  />
</div>
`;

exports[`Modal component Should close the modal on ESC key press 1`] = `
<div
  class="modal is-active"
>
  <div
    class="modal-background"
    role="presentation"
  />
  <div
    class="modal-card"
  >
    <header
      class="modal-card-head"
    >
      <p
        class="modal-card-title"
      >
        Modal Title
      </p>
      <a
        class="delete"
        href=""
      />
    </header>
  </div>
</div>
`;

exports[`Modal component Should expose Content and Card's types 1`] = `[Function]`;

exports[`Modal component Should expose Content and Card's types 2`] = `[Function]`;

exports[`Modal component Should expose Content and Card's types 3`] = `[Function]`;

exports[`Modal component Should expose Content and Card's types 4`] = `[Function]`;

exports[`Modal component Should expose Content and Card's types 5`] = `[Function]`;

exports[`Modal component Should expose Content and Card's types 6`] = `[Function]`;

exports[`Modal component Should no try to reopen if other prop change 1`] = `
<div
  class="modal is-active"
>
  <div
    class="modal-background"
    role="presentation"
  />
  <div
    class="modal-content"
  >
    Content
  </div>
  <button
    aria-label="close"
    class="modal-close is-large"
  />
</div>
`;

exports[`Modal component Should not close the modal if clicked on background 1`] = `
<div
  class="modal is-active"
>
  <div
    class="modal-background"
    role="presentation"
  />
  <div
    class="modal-content"
  >
    Content
  </div>
  <button
    aria-label="close"
    class="modal-close is-large"
  />
</div>
`;

exports[`Modal component Should not close the modal if clicked on background 2`] = `
<div
  class="modal is-active"
>
  <div
    class="modal-background"
    role="presentation"
  />
  <div
    class="modal-content"
  >
    Content
  </div>
  <button
    aria-label="close"
    class="modal-close is-large"
  />
</div>
`;

exports[`Modal component Should not close the modal on ESC key press 1`] = `
<div
  class="modal is-active"
>
  <div
    class="modal-background"
    role="presentation"
  />
  <div
    class="modal-card"
  >
    <header
      class="modal-card-head"
    >
      <p
        class="modal-card-title"
      >
        Modal Title
      </p>
      <a
        class="delete"
        href=""
      />
    </header>
  </div>
</div>
`;

exports[`Modal component Should not close the modal on other than ESC key press 1`] = `
<div
  class="modal is-active"
>
  <div
    class="modal-background"
    role="presentation"
  />
  <div
    class="modal-card"
  >
    <header
      class="modal-card-head"
    >
      <p
        class="modal-card-title"
      >
        Modal Title
      </p>
      <a
        class="delete"
        href=""
      />
    </header>
    <section
      class="modal-card-body"
    >
      Body
    </section>
    <footer
      class="modal-card-foot"
    >
      Footer
    </footer>
  </div>
</div>
`;

exports[`Modal component Should not recreate DOM element to mount Modal if the element is passed down via props 1`] = `
<div
  class="modal is-active"
>
  <div
    class="modal-background"
    role="presentation"
  />
  <div
    class="modal-content"
  >
    Content
  </div>
  <button
    aria-label="close"
    class="modal-close is-large"
  />
</div>
`;

exports[`Modal component Should not show close button 1`] = `
<div
  class="modal is-active"
>
  <div
    class="modal-background"
    role="presentation"
  />
  <div
    class="modal-content"
  >
    Content
  </div>
</div>
`;

exports[`Modal component Should open the modal 1`] = `
<div
  class="modal is-active"
>
  <div
    class="modal-background"
    role="presentation"
  />
  <div
    class="modal-card"
  >
    <header
      class="modal-card-head"
    >
      <p
        class="modal-card-title"
      >
        Modal Title
      </p>
      <a
        class="delete"
        href=""
      />
    </header>
  </div>
</div>
`;

exports[`Modal component Should render any child type 1`] = `
<div
  class="modal is-active"
>
  <div
    class="modal-background"
    role="presentation"
  />
  <div>
    CHILDREN
  </div>
  <div>
    CHILDREN
  </div>
  <button
    aria-label="close"
    class="modal-close is-large"
  />
</div>
`;
